indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Encoding.kt

```diff
@@ -0,0 +1,59 @@
+package at.asitplus.signum.indispensable.asn1
+
+import kotlinx.io.*
+import kotlinx.io.unsafe.UnsafeBufferOperations
+
+@OptIn(UnsafeIoApi::class)
+fun ByteArray.wrapInUnsafeSource(): Source = Buffer().apply {
+    UnsafeBufferOperations.moveToTail(this, this@wrapInUnsafeSource)
+}
+
+/**
+ * Directly moves the byte array to a buffer without copying. Thus, it keeps bytes managed by a Buffer accessible.
+ * The bytes may be overwritten through the Buffer or even recycled to be used by another buffer.
+ * Therefore, operating on these bytes after wrapping leads to undefined behaviour.
+ * [startIndex] is inclusive, [endIndex] is exclusive.
+ */
+@OptIn(UnsafeIoApi::class)
+internal fun wrapInUnsafeSource(bytes: ByteArray, startIndex: Int = 0, endIndex: Int = bytes.size) = Buffer().apply {
+    require(startIndex in 0..endIndex) { "StartIndex bust be between 0 and $endIndex" }
+    UnsafeBufferOperations.moveToTail(this, bytes, startIndex, endIndex)
+}
+
+/**
+ * Helper to create a buffer, operate on it and return its contents as a [ByteArray]
+ */
+inline fun throughBuffer(operation: (Buffer) -> Unit): ByteArray =
```

Comment:
`internal`?

Reply:
required in other modules

Reply:
copied to other module and made internal, as above